src/utilities/use-command.ts

```diff
@@ -0,0 +1,20 @@
+import { COMMANDS } from '../commands/index.js';
+import type { Command } from '../types/command.js';
+import type { CommandConstructorParameters } from '../types/command-constructor-parameters.js';
+import type { CommandDerived } from '../types/command-derived.js';
+
+export function useCommand(
+  commandOrCommandName: CommandDerived | string,
+  params: CommandConstructorParameters
+): Command {
+  const commandDerived =
+    typeof commandOrCommandName === 'string'
+      ? COMMANDS.find((cmd) => new cmd(...params).name === commandOrCommandName)
+      : commandOrCommandName;
+
+  if (!commandDerived) {
+    throw new Error(`Command does not exist`);
```

Comment:
This is a case that shouldn't ever happen unless somehow we don't register commands correctly on start right?

Reply:
That's correct. Its a technically correct check since the type system says its possible it doesn't exist (which is true) but it will never practically happen